tutorials/cloud-pubsub-drainer/index.md

```diff
@@ -0,0 +1,186 @@
+        --trigger-topic drain-tasks \
+        --update-env-vars BUCKET_NAME=$PROJECT_ID-data-archive,SUBSCRIPTION_NAME=bulk-drainer,AUTH_TOKEN=abcd
+
+How much is consolidated per object is configured in the code, but can be set to a number of messages or a certain number of bytes.
+
+### Deploy webhook relay
+
+Stackdriver does not currently have a native Cloud Pub/Sub alert notification channel. We use a small HTTP Cloud Function to act as a simple relay. The `AUTH_TOKEN` environment variable sets an expected shared secret between the Stackdriver Notification Channel and the webhook.
+
+	gcloud functions deploy StackDriverRelay --runtime go111 --trigger-http --update-env-vars AUTH_TOKEN=abcd
+
+### Allow Stackdriver to call the webhook
+
+*Cloud Function Alpha Users Only* 
+
+Stackdriver can only reach publicly accesible webhooks, and if you are using a project with function authorization enabled, you will need to make it reachable.
+
+	gcloud alpha functions add-iam-policy-binding StackDriverRelay --member "allUsers" --role "roles/cloudfunctions.invoker"
+
+### Create Stackdriver Notification Channel
+
+In Stackdriver, a Notification Channel is a resource that can be used with one or more Alert Policies. These can be created in the console or via API.  Using the URL from our deployed function, you create a new channel, and then get the Stackdriver-assigned channel identifier as an environment variable.
+
+	export URL=$(gcloud functions describe StackDriverRelay --format='value(httpsTrigger.url)')
+
+	gcloud alpha monitoring channels create --channel-content-from-file channel.json --channel-labels url=$URL?token=abcd
+
+	export CHANNEL=$(gcloud alpha monitoring channels list --filter='displayName="Archiver"' --format='value("name")')
+
+### Create a Stackdriver Alerting Policy
+
+Alerting Policies describe conditions that result in the alert firing as true. This is where we want to create an alert condition if messages in our "bulk-drainer" Cloud Pub/Sub subscription get either too numerous, or too old.  See the contents of `policy.json` in the tutorial repo for details. The duration and size thresholds are set low so that the solution can be tested and demonstrated quickly.
+
+After the policy is created, the ID is retrieved and then it is updated with to use the Notification Channel created earlier.
+
+	gcloud alpha monitoring policies create --policy-from-file=policy.json
+
+	export POLICY=$(gcloud alpha monitoring policies list --filter='displayName="archive-pubsub"' --format='value("name")')
+
+	gcloud alpha monitoring policies update $POLICY --add-notification-channels=$CHANNEL
+
+	At this point, the solution is fully deployed. You can see and review the policy in the alerting section of Stackdriver Console. You can see notification channels in the workspace settings area of the Stackdriver console.
+
+## Testing the solution
+
+### Create test data
+
+The loader script creates some synthetic test data
+
+	cd ../loader
+	go run main.go
```

Comment:
I had to run a go get command because running without generated the error below.

$ go run main.go
/google/gopath/src/cloud.google.com/go/iam/iam.go:29:2: cannot find package "github.com/googleapis/gax-go/v2" in any of:
        /usr/local/go/src/github.com/googleapis/gax-go/v2 (from $GOROOT)
        /home/cbaer/gopath/src/github.com/googleapis/gax-go/v2 (from $GOPATH)
        /google/gopath/src/github.com/googleapis/gax-go/v2
/google/gopath/src/cloud.google.com/go/pubsub/trace.go:22:2: cannot find package "go.opencensus.io/plugin/ocgrpc" in any of:

Reply:
@ptone , is this comment from Charles something that needs to be resolved before we publish?

Reply:
I don't believe this is an issue if running on cloud shell, where did you run into this @charlesbaer ?

Reply:
Yes, I ran from Cloud Shell.

Reply:
Todd - let me re-run and reproduce. This is something that has gotten me before, since cloud shell setup by default will be preserved across sessions and across projects.

Reply:
@ToddKopriva good to go